scripts/calculate_curtailment.py

```diff
@@ -21,39 +17,41 @@
 from lib.data import *
 
 
-if __name__ == "__main__":
-
-    DATA_DIR = Path(__file__).parent.parent / "data"
-    SAVE_DIR = DATA_DIR / "PHYBM/raw"
-
-    df_bm_units = pd.read_excel(DATA_DIR / "BMUFuelType.xls", header=0)
-
-    df = fetch_physical_data(
-        start_date="2022-03-19 13:00", end_date="2022-03-19 13:30", save_dir=SAVE_DIR
-    )
-    df = format_physical_data(df)
-    df = add_bm_unit_type(df, df_bm_units=df_bm_units)
-
-    df_fpn, df_boal = parse_fpn_from_physical_data(df), parse_boal_from_physical_data(
-        df
-    )
-
-    wind_units = df_boal[df_boal["Fuel Type"] == "WIND"].index.unique()
+def run(db: DbRepository, start_time, end_time):
+    """Fetch data from the DB between `start_time` and `end_time`, and calculate and plot the FPN vs the level
+    specified by the BOAL.
+    """
 
+    df_fpn, df_boal = db.get_data_for_time_range(start_time=start_time, end_time=end_time)
     curtailment_dfs = []
+    units = df_boal.index.unique()
 
-    for unit in wind_units:
-        df_curtailment_unit = analyze_one_unit(
-            df_boal_unit=df_boal.loc[unit], df_fpn_unit=df_fpn.loc[unit]
-        )
+    for unit in units:
+        df_curtailment_unit = analyze_one_unit(df_boal_unit=df_boal.loc[unit], df_fpn_unit=df_fpn.loc[unit])
 
         curtailment_in_mwh = calculate_curtailment_in_mwh(df_curtailment_unit)
         generation_in_mwh = calculate_notified_generation_in_mwh(df_curtailment_unit)
 
-        print(
-            f"Curtailment for {unit} is {curtailment_in_mwh:.2f} MWh. Generation was {generation_in_mwh:.2f} MWh"
-        )
+        print(f"Curtailment for {unit} is {curtailment_in_mwh:.2f} MWh. Generation was {generation_in_mwh:.2f} MWh")
 
         curtailment_dfs.append(df_curtailment_unit)
 
     df_curtailment = pd.concat(curtailment_dfs)
+    total_curtailment = df_curtailment["delta"].sum() * MINUTES_TO_HOURS
+    print(f"Total curtailment was {total_curtailment:.2f} MWh ")
+
+    df_ = df_curtailment.reset_index().groupby(["Time"]).sum().reset_index()
+
+    fig = px.area(df_, x="Time", y=["Level_FPN", "Level_After_BOAL"])
+    fig.update_traces(stackgroup=None, fill="tozeroy")
+    fig.update_layout(
+        yaxis=dict(title="MW"),
+        title=dict(text=f"Total Curtailment {total_curtailment:.2f} MWh"),
+    )
+
+    fig.show()
+
+
+if __name__ == "__main__":
```

Comment:
I'd be always tempted to move this to a test, and run this with CI, but I know this is good to get it going

Reply:
but this is actually running the script, not testing it!

Reply:
ah ignore it then, 

Perhaps I'm just always so happy to get things tested, so would be to move it this away from scripts and then have a simple test, running one day. This perhaps can be a done later on, as its not urgent